be1-go/internal/handler/channel/federation/hfederation/federation.go

```diff
@@ -366,19 +378,28 @@ func (h *Handler) handleChallenge(msg mmessage.Message, channelPath string,
 		// called => broadcast the result to both federation channels directly.
 		_ = h.db.StoreMessageAndData(remoteChannel, resultMsg)
 		_ = h.subs.BroadcastToAllClients(resultMsg, remoteChannel)
+
+		h.log.Info().Msgf("A federation was created with the local LAO %s",
+			federationExpect.LaoId)
 	} else {
+		// Add the socket to the list of server sockets
+		h.sockets.Upsert(socket)
+
+		// Send the rumor state directly to avoid delay while syncing
+		err = h.rumors.SendRumorStateTo(socket)
+		if err != nil {
+			return err
+		}
```

Comment:
When a server connect to an other server, the receiving server already notify the hub of a new socket and will already upsert the socket and send the rumor state message so you don't need to do it here

Reply:
as for now, the federation protocol is using the client socket, therefore we need to do that